rcgen.js

```diff
@@ -0,0 +1,88 @@
+// @ts-check
+
+const {gitFiles} = require('./rcgen/git/files');
+const {ignoreFilesWithGit} = require('./rcgen/git/utils');
+const {prettierFiles} = require('./rcgen/prettier/files');
+const {prettierPatchers} = require('./rcgen/prettier/patchers');
+const {
+  configurePrettier,
+  ignoreFilesWithPrettier
+} = require('./rcgen/prettier/utils');
+const {vscodeFiles} = require('./rcgen/vscode/files');
+const {vscodePatchers} = require('./rcgen/vscode/patchers');
+const {
+  excludeFilesFromSearchInVscode,
+  excludeFilesInVscode,
+  recommendVscodeExtensions
+} = require('./rcgen/vscode/utils');
+
+/**
+ * @type {import('@rcgen/core').Manifest}
+ */
+exports.default = {
+  files: [...gitFiles, ...prettierFiles, ...vscodeFiles],
+  patchers: [
+    ...prettierPatchers,
+    ...vscodePatchers,
+    ignoreFilesWithGit(
+      '.cache',
+      'coverage',
+      'lerna-debug.log',
+      'lib',
+      'node_modules',
+      'npm-debug.log',
+      'package-lock.json',
+      'packages/website/build',
+      'packages/website/i18n',
+      'todo.tasks',
+      'yarn-error.log'
+    ),
+    configurePrettier({
```

Comment:
are the prettier settings typesafe here or do I lose the intellisense I would have in the regular config file provided through the public json schema?

Reply:
i will try to combine json-schema and typescript for this issue